### PR TITLE
Update sync documentation to explain new system

### PR DESCRIPTION
**Pull request by jfy133**
This adds references to the 'working' copy of TEMPLATE that is opened against `dev` when an automated sync occurs.

### CODE REVIEW

File: markdown/developers/sync.md [hunk 1]
```diff
@@ -25,8 +25,8 @@ These pull requests then need to be manually resolved and merged by the pipeline
 Behind the scenes, this synchronisation is done by using `git`.
 Each repository has a special `TEMPLATE` branch which contains only the "vanilla" code made by the `nf-core create` tool.
 The synchronisation tool fetches the essential variables needed to recreate the pipeline and uses this to trigger a `nf-core create --no-git` command with the latest version of the template.
-The result from this is then compared against what is stored in the `TEMPLATE` branch and committed.
-When merging from the `TEMPLATE` branch back into the main `dev` branch of the pipeline, `git` should be clever enough to know what has changed since the template was first used, and therefore, it will only present the relevant changes.
+The result from this is then compared against what is stored in the `TEMPLATE` branch and committed. During an automated sync, a copy of the `TEMPLATE` branch will made called `nf-core-template-merge-<version>`, and a PR from this new branch will be opened against your `dev`.
```
Comment (alneberg) [inline markdown/developers/sync.md:28]:
```suggestion
The result from this is then compared against what is stored in the `TEMPLATE` branch and committed. During an automated sync, a copy of the `TEMPLATE` branch called `nf-core-template-merge-<version>` will be made, and a PR from this new branch will be opened against your `dev`.  
```

**Review by alneberg — APPROVED** [review]
Looks good to me! Made a suggestion where a word seemed to be missing.

**Comment by apeltzer** [comment]
I think this is good 👍🏻

**Review by apeltzer — APPROVED** [review]

File: markdown/developers/sync.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ Sorry about this, but there's no way around it..
 You should not be actively working on the main nf-core repository, so we need to bring these changes to your personal fork.
```
Comment (ewels) [inline markdown/developers/sync.md:46]: Can maybe make this a little softer now:

```
You can either work on the branch created for the template sync to fix the merge conflicts or pull the updates to `TEMPLATE` to your own branch.
Working on your fork is recommended if the merge is not trivial (please make a comment on the automated PR to say that you are working on it though).

```

File: markdown/developers/sync.md [hunk 3]
```diff
@@ -46,7 +46,7 @@ Sorry about this, but there's no way around it..
...
 The steps we need to do are:
 
-1. Pull the `nf-core/<pipeline>` `TEMPLATE` changes to your fork
+1. Pull the `nf-core/<pipeline>` `TEMPLATE` changes to your fork (or the `nf-core-template-merge-<version>` branch)
```
Comment (ewels) [inline markdown/developers/sync.md:49]: Should be identical, so no need to complicate things by mentioning the branch here I think.

Could be good to give example commands here though:

```bash
git remote add upstream <url>
git fetch upstream
git checkout -b merge-template-updates
git merge upstream/TEMPLATE
# [..fix merge conflicts, git add changes..]
git commit -m "Fixed merge conflicts"
git push --set-upstream origin merge-template-updates
# Open pull-request from your fork
```

Comment (ewels) [reply markdown/developers/sync.md:49]: ah fail, this is already in the document just below 🤦🏻

File: markdown/developers/sync.md [hunk 4]
```diff
@@ -69,7 +69,7 @@ Next, check out a new branch to make these changes in:
 git checkout -b merging-template-updates
 ```
 
-Finally, pull the `TEMPLATE` branch from the `upstream` repo:
+Finally, pull the `TEMPLATE` (or `nf-core-template-merge-<version>`) branch from the `upstream` repo:
```
Comment (ewels) [inline markdown/developers/sync.md:72]: No need for this change, I think it just complicates matters when pulling to a fork - the two should be identical.

File: markdown/developers/sync.md (hunk 2 above)
Comment (ewels) [reply markdown/developers/sync.md:46]: Maybe also restructure / rewrite a little to add a paragraph about how simple merge conflicts can be fixed in the GitHub PR interface?

**Pull request merged**